### PR TITLE
[WIP] Login rate limit: no-cache implementation

### PR DESCRIPTION
**Pull request by bleistivt**
Currently login rate limiting is limited to installs with caching. This
tries to fix that.

WIP because I need feedback on how to implement IP rate limiting without
being too DB-expensive.

The actual fix is this part:
https://github.com/bleistivt/vanilla/blob/a2d656b136cd86b5a59a48f3249dc2896e5da7c1/applications/dashboard/models/class.usermodel.php#L3458-L3474

### CODE REVIEW

File: applications/dashboard/models/class.usermodel.php [hunk 1]
```diff
@@ -3454,75 +3454,113 @@ public function PasswordReset($UserID, $Password) {
     * @param array $User
     * @param boolean $PasswordOK
     */
-   public static function RateLimit($User, $PasswordOK) {
-      if (!Gdn::Cache()->ActiveEnabled()) return FALSE;
-//      $CoolingDown = FALSE;
+    public static function RateLimit($User, $PasswordOK) {
+        if (function_exists('apc_store')) {
```
Comment (tburry) [inline applications/dashboard/models/class.usermodel.php]: The check to `Gdn::Cache()->ActiveEnabled()` must come first. On multiple application server environments you need a global check.

Comment (bleistivt) [inline applications/dashboard/models/class.usermodel.php]: Ok, so I would rewrite it to be:

    if (Gdn::Cache()->ActiveEnabled())
        // Use cache
    elseif (C('Garden.Apc', false) && function_exists('apc_store'))
        // Use apc
    else
        // Use database (user attributes)

What do you think about  removing the large comment block of dead code?

Comment (tburry) [inline applications/dashboard/models/class.usermodel.php]: Yes please with braces on top.

**Comment by bleistivt** [comment]
new pull to current stage #

**Pull request closed without merging**